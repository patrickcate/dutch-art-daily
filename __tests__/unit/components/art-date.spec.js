import { shallowMount } from '@vue/test-utils'
import ArtDate from '@components/art-date.vue'

describe('ArtDate Component', () => {
  it('should be a Vue instance', () => {
    const wrapper = shallowMount(ArtDate, {
      mocks: {
        $store: {
          state: {
            currentPage: '01-01',
          },
        },
      },
    })

    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('renders correctly', () => {
    process.client = true

    const wrapper = shallowMount(ArtDate, {
      mocks: {
        $store: {
          state: {
            currentPage: '01-01',
          },
        },
      },
    })

    expect(wrapper).toMatchSnapshot()
  })

  it('does not call set setBoundingBox() if no child elements', () => {
    const setBoundingBoxMock = jest.fn()

    const wrapper = shallowMount(ArtDate, {
      mocks: {
        $store: {
          state: {
            currentPage: '01-01',
          },
        },
        setBoundingBox() {
          setBoundingBoxMock()
        },
      },
    })

    wrapper.vm.$children = false

    expect(setBoundingBoxMock).not.toHaveBeenCalled()
  })

  // TODO: Test that changing date renders new date correctly
  // TODO: Test style top/left are correcly added.
})
