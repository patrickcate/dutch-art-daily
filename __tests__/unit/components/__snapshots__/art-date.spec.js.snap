// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtDate Component renders correctly 1`] = `
<transition-group-stub tag="div" name="date" type="transition" class="artwork-date">
  <!----> <span class="artwork-date__item">January&nbsp;</span> <span class="artwork-date__item artwork-date__char">1</span> <sup class="artwork-date__item artwork-date__ordinal">
    st
  </sup></transition-group-stub>
`;
